### PR TITLE
added magnetometer calibration computation scripts.

### PR DESCRIPTION
**Pull request by shokoofeh**

### CODE REVIEW

File: jackal_base/launch/base.launch [hunk 1]
```diff
@@ -24,11 +32,13 @@
   <!-- Filter raw gyro data into a usable IMU message -->
   <node pkg="imu_filter_madgwick" type="imu_filter_node" name="imu_filter">
     <rosparam>
-      gain: 0.005
-      use_mag: false
+      gain: 0.1
```
Comment (mikepurvis) [inline jackal_base/launch/base.launch:29]: Didn't we have issues with the higher gain values? Is this okay?

File: jackal_base/launch/base.launch [hunk 2]
```diff
@@ -1,5 +1,13 @@
...
-  <include file="$(find jackal_description)/launch/description.launch" />
+
+  <include file="$(find jackal_description)/launch/description.launch" >
+    <arg name="front_laser" value="true"/>
+    <arg name="accessories" value="true" />
+  </include>
+
+  <include file="$(find lms1xx)/launch/LMS1xx.launch">
+    <arg name="frame_id" value="front_laser"/>
+  </include>
```
Comment (mikepurvis) [inline jackal_base/launch/base.launch]: Please exclude these changes.

File: jackal_base/scripts/calibrate_compass [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+TEMPDIR=$(mktemp -d --tmpdir calibrate_compass.XXXX)
+BAG_FILE=${TEMPDIR}/imu_record.bag
+CAL_FILE=${TEMPDIR}/imu_compass.yaml
+DURATION=60
+
+ROSTOPIC_OUTPUT=$(rostopic list)
+if [[ "$?" != "0" ]]; then
+  echo "ROS appears not to be running. Please start ROS service:"
+  echo "sudo service jackal-core start"
```
Comment (mikepurvis) [inline jackal_base/scripts/calibrate_compass]: On the production bots, the service is actually just called `ros`. So this should be:

```
sudo service ros start
```

File: jackal_base/scripts/compute_calibration [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+  exit(1)
+
+# Plots are optional
+try:
+  from matplotlib import pyplot
+  from mpl_toolkits.mplot3d import Axes3D
+except ImportError:
+  pyplot = None
+
+parser = ArgumentParser(description='Process UM6 bag file for compass calibration. Pass a bag containing /imu/rpy/raw and /imu/mag topics, with the UM6 compass facing upright, being slowly rotated in a clockwise direction for 30-120 seconds.')
```
Comment (mikepurvis) [inline jackal_base/scripts/compute_calibration]: Remove UM6 reference here, please.

File: jackal_base/scripts/compute_calibration [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+import rosbag
+
+import datetime
+from tf.msg import tfMessage
+from argparse import ArgumentParser
+from geometry_msgs.msg import Quaternion
+from numpy import mean, array, hypot, diff, convolve, arange, sin, cos, ones, pi, matrix
+from tf.transformations import euler_from_quaternion,quaternion_from_euler,quaternion_multiply,quaternion_matrix
+import tf
```
Comment (mikepurvis) [inline jackal_base/scripts/compute_calibration:11]: All these imports are going to necessitate some extra `run_depends` in `package.xml`.

File: jackal_base/launch/base.launch (hunk 1 above)
Comment (shokoofeh) [inline jackal_base/launch/base.launch:29]: It is increased to the default value since the convergence was very slow.

**Comment by shokoofeh** [comment]
changes has been made: 0ca4b3ba73
I'm not sure if I add all neccessary dependencies!

File: jackal_base/launch/base.launch (hunk 1 above)
Comment (mikepurvis) [inline jackal_base/launch/base.launch:29]: Okay, sounds good.

**Comment by mikepurvis** [comment]
LGTM, thanks.

**Pull request merged**